### PR TITLE
[FS] Store deposits in a neat little repository

### PR DESCRIPTION
**Pull request by codingupastorm**
Note that this is going to break the federated peg code for now, but gives us a much nicer start to build off of

### CODE REVIEW

File: src/Stratis.Features.FederatedPeg/SourceChain/Deposit.cs [hunk 1]
```diff
@@ -4,36 +4,62 @@
...
+            this.id = id;
+            this.amount = amount;
+            this.targetAddress = targetAddress;
+            this.blockNumber = blockNumber;
+            this.blockHash = blockHash;
         }
 
         /// <inheritdoc />
-        public uint256 Id { get; }
+        public uint256 Id { get { return this.id; } }
```
Comment (rowandh) [inline src/Stratis.Features.FederatedPeg/SourceChain/Deposit.cs]: =>

File: src/Stratis.Features.FederatedPeg.Tests/DepositRepositoryTests.cs [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+
+namespace Stratis.Features.FederatedPeg.Tests
+{
+    public class DepositRepositoryTests
+    {
+        private readonly DepositRepository depositRepository;
+
+        public DepositRepositoryTests()
+        {
+            var network = new CirrusRegTest();
```
Comment (rowandh) [inline src/Stratis.Features.FederatedPeg.Tests/DepositRepositoryTests.cs]: `CirrusNetwork.NetworksSelector.Regtest`

File: src/Stratis.Sidechains.Networks/CirrusV2/CirrusRegTest.cs [hunk 3]
```diff
@@ -28,7 +28,7 @@ public class CirrusRegTest : PoANetwork, ISignedCodePubKeyHolder
 
         public PubKey SigningContractPubKey { get; }
 
-        internal CirrusRegTest()
+        public CirrusRegTest()
```
Comment (rowandh) [inline src/Stratis.Sidechains.Networks/CirrusV2/CirrusRegTest.cs]: Can remain `internal`

File: src/Stratis.Features.FederatedPeg/TargetChain/DepositRepository.cs [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+
+            this.serializer = serializer;
+        }
+
+        /// <inheritdoc />
+        public int GetSyncedBlockNumber()
+        {
+            using (Transaction dbreezeTransaction = this.db.GetTransaction())
+            {
+                Row<byte[], int> row = dbreezeTransaction.Select<byte[], int>(DepositTableName, SyncedUpToBlockKey);
```
Comment (rowandh) [inline src/Stratis.Features.FederatedPeg/TargetChain/DepositRepository.cs]: I don't see anywhere where the value of `SyncedUpToBlockKey` is updated.

File: src/Stratis.Features.FederatedPeg/TargetChain/MaturedBlocksSyncManager.cs [hunk 5]
```diff
@@ -90,55 +80,29 @@ private async Task RequestMaturedBlocksContinouslyAsync()
...
-                else
-                {
-                    this.logger.LogDebug("Considering ourselves fully synced since no blocks were received");
-
-                    // If we've received nothing we assume we are at the tip and should flush.
-                    // Same mechanic as with syncing headers protocol.
-                    await this.store.SaveCurrentTipAsync().ConfigureAwait(false);
-                }
+                // We found new deposits. Add them to the store.
+                this.depositRepository.SaveDeposits(matureBlockDeposits);
```
Comment (rowandh) [inline src/Stratis.Features.FederatedPeg/TargetChain/MaturedBlocksSyncManager.cs]: This method now only returns `true`.

File: src/Stratis.Features.FederatedPeg/TargetChain/MaturedBlocksSyncManager.cs [hunk 6]
```diff
@@ -90,55 +80,29 @@ private async Task RequestMaturedBlocksContinouslyAsync()
...
             }
 
-            return delayRequired;
+            // TODO: Do we need to update the most recently highest block
+
+            return true;
```
Comment (rowandh) [inline src/Stratis.Features.FederatedPeg/TargetChain/MaturedBlocksSyncManager.cs]: This method now only returns `true`.

File: src/Stratis.Features.FederatedPeg/SourceChain/Deposit.cs (hunk 1 above)
Comment (codingupastorm) [reply src/Stratis.Features.FederatedPeg/SourceChain/Deposit.cs]:
```suggestion
        public uint256 Id => this.id;
```

Comment (rowandh) [inline src/Stratis.Features.FederatedPeg/SourceChain/Deposit.cs]:
```suggestion
        public uint256 Id => this.id;
```

**Pull request closed without merging**